compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/ProvidesMethodFactoryGenerator.kt

```diff
@@ -338,4 +339,36 @@ internal class ProvidesMethodFactoryGenerator : PrivateCodeGenerator() {
       name
     }
   }
+
+  // TODO: May be worth extracting some of this to an interface to actually apply to the Reference
+  // classes if we think it would be useful in other use-cases
```

Comment:
When I looked at this I thought the same. I was thinking of introducing a `CallableReference` interface and make `FunctionReference` and `PropertyReference` implement this interface. 

On the other hand, it is only used for the `ProvidesMethodFactoryGenerator`. So maybe we could keep this private to this class for now? 

What do you think?

Reply:
Since there aren't any other use-cases yet, I'm ok keeping it as just the private class for now. However, I think it's worth us looking at what fields/methods are common across all the different Reference types and potentially extracting a generic interface for those. E.g. it seems all References will have annotations, fqName, and visibility. I also figured if we do put any of these in interfaces, it would make sense to do it separately once you're changes are in as well to minimize the conflicts :)